parser/lexer.go

```diff
@@ -435,7 +470,16 @@ func scanIdentifier(s *Scanner) (int, Pos, string) {
 	pos := s.r.pos()
 	s.r.inc()
 	s.r.incAsLongAs(isIdentChar)
-	return identifier, pos, s.r.data(&pos)
+	lit := s.r.data(&pos)
```

Comment:
It's quite performance sensitive here.
`strings.ToLower(lit)` would cause allocation, and map access is not that fast.
I think you can switch the order, check IgnoreSpaceMode first, try the best not to hurt the performance in most cases.

Reply:
@tiancaiamao ok, I'll take a look. thx :)

Reply:
@tiancaiamao I've refine this place. PTAL